state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -1006,7 +936,239 @@ fn witness_broadcast(tx_out_id: [u8; 4]) {
+				},
+			));
+		})
+		.then_execute_at_next_block(|_| {})
+		.then_execute_with(|_| {
+			BroadcastDelay::set(Some(10));
+			// Set delay - retries will be added to the Delayed queue.
+			assert_ok!(Broadcaster::transaction_failed(RuntimeOrigin::signed(1u64), broadcast_id));
+			target = System::block_number() + 10;
+
+			assert_eq!(BroadcastRetryQueue::<Test, Instance1>::decode_len(), Some(0));
+			assert_eq!(
+				DelayedBroadcastRetryQueue::<Test, Instance1>::get(target)[0].broadcast_id,
+				broadcast_id
+			);
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastRetryScheduled {
+					broadcast_id,
+					retry_block: target,
+				},
+			));
+		})
+		.then_execute_at_block(target, |_| {})
+		.then_execute_with(|_| {
+			assert_eq!(BroadcastRetryQueue::<Test, Instance1>::decode_len(), Some(0));
+			assert_eq!(DelayedBroadcastRetryQueue::<Test, Instance1>::decode_len(target), None);
+
+			assert_transaction_broadcast_request_event(broadcast_id, tx_out_id);
+		});
+}
+
+#[test]
+fn broadcast_timeout_delay_works() {
+	let mut target = 0;
+	let mut broadcast_id = 0;
+	new_test_ext()
+		.execute_with(|| {
+			(broadcast_id, _) = start_mock_broadcast();
+
+			BroadcastDelay::set(Some(10));
+			target = System::block_number() + BROADCAST_EXPIRY_BLOCKS;
+		})
+		.then_execute_at_block(target, |_| {})
+		.then_execute_with(|_| {
+			target = System::block_number() + 10;
+
+			assert_eq!(BroadcastRetryQueue::<Test, Instance1>::decode_len(), Some(0));
+			assert_eq!(
+				DelayedBroadcastRetryQueue::<Test, Instance1>::get(target)[0].broadcast_id,
+				broadcast_id
+			);
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastRetryScheduled {
+					broadcast_id,
+					retry_block: target,
+				},
+			));
+		});
+}
+
+#[test]
+fn aborted_broadcasts_will_not_retry() {
+	let mut target = 0;
+	let mut broadcast_id = 0;
+	new_test_ext()
+		.execute_with(|| {
+			(broadcast_id, _) = start_mock_broadcast();
+			BroadcastDelay::set(Some(100));
+			target = System::block_number() + 100;
+			assert_ok!(Broadcaster::transaction_failed(RuntimeOrigin::signed(0u64), broadcast_id));
+			assert_eq!(
+				DelayedBroadcastRetryQueue::<Test, Instance1>::get(target)[0].broadcast_id,
+				broadcast_id
+			);
+
+			// Abort the broadcast
+			let nominee = ready_to_abort_broadcast(broadcast_id);
+			assert_ok!(Broadcaster::transaction_failed(
+				RuntimeOrigin::signed(nominee),
+				broadcast_id
+			));
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastAborted { broadcast_id },
+			));
+		})
+		.then_execute_at_block(target, |_| {})
+		.then_execute_with(|_| {
+			// assert no retry happened
+			assert!(FailedBroadcasters::<Test, Instance1>::get(broadcast_id).is_empty());
+			assert!(!PendingBroadcasts::<Test, Instance1>::get().contains(&broadcast_id))
+		});
+}
+
+#[test]
+fn succeeded_broadcasts_will_not_retry() {
```

Comment:
what if there's a `transaction_failure` after success (again could leave this to another PR)

Reply:
Added this case to the unit test